---
title: Molecular Storage and Computing

description: Working at the intersection of biology and architecture, including biologically-inspired computer systems like DNA storage, hybrid electronic-molecular systems, and next-generation bioinformatics applications.

people:
  - kendall
  - mwillsey
  - luisceze
  - karin

layout: project
link: http://misl.cs.washington.edu/
---

## DNA Storage
The installed capacity of archival storage has not kept up with exponential
growth in the global production of data, creating a storage gap that cannot be
filled by conventional technology. DNA has an upper density bound of 1
exabyte/mm<sup>3</sup>, making it a promising candidate for a next-generation
archival storage medium.

In collaboration with the [Molecular Information Systems Group](
http://misl.cs.washington.edu/) at UW, our research encompasses research into
encoding schemes and system designs that will support practical
implementations of this emerging technology.

### Publications:
[A DNA-Based Archival Storage System](
http://homes.cs.washington.edu/~bornholt/papers/dnastorage-asplos16.pdf). <br>
James Bornholt, Randolph Lopez, Douglas M. Carmean, Luis Ceze, Georg Seelig, and
Karin Strauss. <br>
International Conference on Architectural Support for Programming Languages and
Operating Systems (ASPLOS), April 2016.


## ACGT: Accelerating Genomics Tools
Personalized whole genome sequencing is an emerging technology that has the potential to have a profound impact on global health.
However, current state-of-the art DNA sequencing and alignment algorithms require tremendous amounts of compute and memory resources which present significant challenges for deploying this technology at scale.
In order to make this technology ubiquitous and portable enough to be easily and widely deployed, substantial gains in computing performance, energy efficiency, and form factor must be made.

At the same time, the deployment and development of heterogeneous computing accelerators, emerging memory substrates, and even custom ASICs have been shown to yield promising gains through specialization.
This project investigates the space of heterogeneous hardware and software solutions to accelerate entire genome-analysis algorithms including sequencing, alignment, and variant-calling.
